### PR TITLE
  [BUG] Fix MLM target tensor cloned from x instead of y in MaskedDat…

### LINKED ISSUE #461: [BUG] MaskedDataset.__getitem__ trains MLM against input x instead of target y
**Describe the bug**

In pyaptamer/datasets/dataclasses/_masked.py, the masked target tensor is cloned from x instead of y:     
                                                            
  ### Before                                                                                          
  y_masked = x.clone().detach()                                                                                 
                                                                                                                
  ### After                                                                                          
  y_masked = y.clone().detach()                                                                                 
  
  Because y_masked was built from x, the Masked Language Model pretraining objective was always comparing       
  predictions against the input sequence rather than the target sequence. For any dataset where x != y (e.g.,
  cross-modal or cross-species pairs), this makes the entire pretraining objective meaningless — the model
  learns to reconstruct its own noisy input, not the intended target. 

**Versions**
0.1.0a1

### PR DESCRIPTION
**Pull request by ritankarsaha**
#### Reference Issues/PRs

Fixes #461 

#### What does this implement/fix? Explain your changes.

Fixed MLM target cloned from wrong tensor in MaskedDataset. MaskedDataset.__getitem__ constructed y_masked by cloning x rather than y, so the MLM loss was computed against the input sequence. This fix changes the clone source to y, aligning the pretraining objective with the intended supervision signal.                                                                              
                                                                                                                
#### Did you add any tests for the change?

Yes tests have been added and successfully passes all checks. 

#### PR checklist

- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.
- [x] Added/modified tests
- [x] Used pre-commit hooks when committing to ensure that code is compliant with hooks. Install hooks with `pre-commit install`.
  To run hooks independent of commit, execute `pre-commit run --all-files`

@NennoMP

### CODE REVIEW

**Comment by ritankarsaha** [comment]
Hey @NennoMP made a PR for the bug, can you please take a look into it ? Thanks !
